### PR TITLE
Updating HostWarmupMiddleware to work with placeholder simulation locally.

### PR DESCRIPTION
**Pull request by kshyju**
Updating HostWarmupMiddleware to enable /warmup API call when testing locally using DebugPlaceholder or ReleasePlaceholder configuration

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

**Review by safihamid — DISMISSED** [review]
:shipit:

**Review by surgupta-msft — APPROVED** [review]

**Comment by surgupta-msft** [comment]
Does this require any documentation changes? PR description says that, but I am not seeing any documentation issue or github issue linked with the PR.

**Comment by kshyju** [comment]
I will push another iteration with some more improvements.

File: src/WebJobs.Script.WebHost/Middleware/HostWarmupMiddleware.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ public class HostWarmupMiddleware
 
         public Task Invoke(HttpContext httpContext)
         {
-            if (IsWarmUpRequest(httpContext.Request, _webHostEnvironment.InStandbyMode, _environment))
+            if (_webHostEnvironment.InStandbyMode)
```
Comment (kshyju) [inline src/WebJobs.Script.WebHost/Middleware/HostWarmupMiddleware.cs:56]: Removed this full "IsWarmupRequest" check as it is redundant. We execute this middleware only if that condition matches - code [here](https://github.com/Azure/azure-functions-host/blob/a4ed4428e75aa6547afb1f6b76d219268e12d613/src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs#L71)

**Comment by kshyju** [comment]
> Does this require any documentation changes? PR description says that, but I am not seeing any documentation issue or github issue linked with the PR.

My bad. No docs change needed. Updated the checklist.

**Review by safihamid — APPROVED** [review]
:shipit:

**Pull request merged**